src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Helpers.cs

```diff
@@ -196,50 +146,70 @@ private static bool IsAccessible(ISymbol symbol)
             Contract.ThrowIfFalse(equivalentTypesWithDifferingAssemblies.All(kvp => kvp.Key.ContainingType == null));
             Contract.ThrowIfFalse(equivalentTypesWithDifferingAssemblies.All(kvp => kvp.Value.ContainingType == null));
 
-            var referencedAssemblies = new MultiDictionary<string, IAssemblySymbol>();
-            foreach (var assembly in compilation.GetReferencedAssemblySymbols())
+            // Cache compilations so we avoid recreating any as we walk the pairs of types.
+            using var _ = PooledHashSet<Compilation>.GetInstance(out var compilationSet);
+
+            foreach (var (type1, type2) in equivalentTypesWithDifferingAssemblies)
             {
-                referencedAssemblies.Add(assembly.Name, assembly);
+                // Check if type1 was forwarded to type2 in type2's compilation, or if type2 was forwarded to type1 in
+                // type1's compilation.  We check both direction as this API is called from higher level comparison APIs
+                // that are unordered.
+                if (!VerifyForwardedType(solution, type1, type2, compilationSet, cancellationToken) &&
+                    !VerifyForwardedType(solution, type2, type1, compilationSet, cancellationToken))
+                {
+                    return false;
+                }
             }
 
-            var verifiedCount = 0;
-            foreach (var kvp in equivalentTypesWithDifferingAssemblies)
+            return true;
+        }
+
+        /// <summary>
+        /// Returns <see langword="true"/> if <paramref name="type1"/> was forwarded to <paramref name="type2"/> in
+        /// <paramref name="type2"/>'s <see cref="Compilation"/>.
+        /// </summary>
+        private static bool VerifyForwardedType(
+            Solution solution,
+            INamedTypeSymbol type1,
+            INamedTypeSymbol type2,
+            HashSet<Compilation> compilationSet,
+            CancellationToken cancellationToken)
+        {
+            // Only need to operate on original definitions.  i.e. List<T> is the type that is forwarded,
+            // not List<string>.
+            type1 = type1.OriginalDefinition;
+            type2 = type2.OriginalDefinition;
+
+            var type2OriginatingProject = solution.GetOriginatingProject(type2);
+            if (type2OriginatingProject == null)
+                return false;
+
+            var type2Compilation = type2OriginatingProject.GetRequiredCompilationAsync(cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
don't love the blocking here.  however, bubblign up async higher proved to be enormous.  So this keeps things very targetted right now.

Reply:
I'm making this asynchronous as a follow-up change, so this review can just defer on the subject. 👍